### PR TITLE
Update goreleaser to v1.2.4 in build image

### PR DESCRIPTION
**Pull request by A-kanksh-a**
## Change Overview
- Use `goreleaser` version 1.2.4
- update build image to version `ghcr.io/kanisterio/build:v0.0.15`
- Update binaries field in `goreleaser.yml` as ids
- Use semantic version tag` v9.99.9-dev` instead of `latest`
- Add Make target `build- controller` to build `controller` binary

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E
Pushed build image with tag `ghcr.io/kanisterio/build:v0.0.15` and respective application images manually with tag `v9.99.9-dev`

### CODE REVIEW

File: examples/kafka/confluent-s3-connector/kafka-blueprint.yaml [hunk 1]
```diff
@@ -139,7 +139,7 @@ actions:
           containers:
           - name: container
             imagePullPolicy: Always
-        image: akankshakumari393/confluent-kafka-s3source:latest
+        image: akankshakumari393/confluent-kafka-s3source:v9.99.9-dev
```
Comment (viveksinghggits) [inline examples/kafka/confluent-s3-connector/kafka-blueprint.yaml:142]: have we been using these images from your personal repo?

Comment (A-kanksh-a) [reply examples/kafka/confluent-s3-connector/kafka-blueprint.yaml:142]: we are not using confluent connectors anymore. For Kafka we are using the adobe sink and source connector.[https://github.com/kanisterio/kanister/tree/master/examples/kafka/adobe-s3-connector](https://github.com/kanisterio/kanister/tree/master/examples/kafka/adobe-s3-connector)

File: .goreleaser.yml [hunk 2]
```diff
@@ -137,3 +137,5 @@ changelog:
     exclude:
     - '^docs:'
     - '^test:'
+archives:
+  - allow_different_binary_count: true
```
Comment (viveksinghggits) [inline .goreleaser.yml:141]: let's add a newline here.

**Review by PrasadG193 — COMMENTED** [review]
Looks good

**Review by PrasadG193 — APPROVED** [review]

**Pull request merged**